switchapi/switch_base_types.h

```diff
@@ -1,18 +1,18 @@
 /*
-Copyright 2013-present Barefoot Networks, Inc.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
+ * Copyright (c) 2022 Intel Corporation.
```

Comment:
This file should probably carry the Barefoot copyright as well as the Intel copyright.
It is a derivative of a previously copyrighted work, not a new work.

Reply:
Since Barefoot is part of Intel now, Namrata mentioned to have Intel and Apache license. Separately Barefoot is not required

Reply:
That's not the way copyright law works.

If we were to treat Barefoot's copyright as having been absorbed by Intel, our notice would read:

`Copyright 2013-2022 Intel Corporation`

The intellectual property content did not originate in 2022, it originated when Barefoot wrote the first line of code (putatively 2013). The practice is to state the year in which you _first_ created intellectual property content, followed by the _last_ year in which you created intellectual copyright content if it is not the same as the first.

Does Namrata have a finding from the Business Unit attorney on this? If not, I think we have to go by Intel policy, which is to treat our copyright as being secondary to Barefoot's original copyright.

Reply:
Legal questions aside, I'm also concerned about appearances. There are copies of the code on _p4lang_ with Barefoot's name and the year 2013 on it. If you don't know that Intel acquired Barefoot, what are you going to think if you see the same three directories of files and a claim that Intel authored it in 2022?